rs/index/src/collection/collection.rs

```diff
@@ -214,6 +234,10 @@ impl<Q: Quantizer + Clone + Send + Sync + 'static> Collection<Q> {
                                                     entry_seq_no,
                                                 )
                                                 .unwrap();
+
+                                            // The doc is not immediately searchable, but we update the metrics anyway
+                                            INTERNAL_METRICS
```

Comment:
this is expensive - don't increment metric every iteration like this. just have a local variable to do that, and then increment the metric at the end